woocommerce/wc-admin
# Add leaderboard section

Fixes #1183 

This PR adds:
* The section header for Leaderboards
* The ellipsis dropdown menu with leaderboard toggles and rows per table option
* User preference persistence for selected leaderboards and rows per table
* Moves the `Leaderboard` to components and creates a separate `Leaderboards` section.

### Accessibility




### Screenshots
<img width="1034" alt="screen shot 2019-01-04 at 3 05 59 pm" src="https://user-images.githubusercontent.com/10561050/50676980-4398b000-1032-11e9-8117-02d744362501.png">

### Detailed test instructions:

1.  Open the dashboard and note the addition of the leaderboard section.
2.  Try toggling the visible leaderboards in the ellipsis menu.  Currently the only one that has been added is "Top Products."
3.  Check that the Top Products leaderboard remains on/off based on your selected preferences after refreshing.
4.  Check that changing rows per table changes the number of visible rows in the leaderboard.
5.  Check that refreshing the page persists rows per table setting.

### Questions
* Same issue as #1230 with user preferences.  User preferences are retrieved async which results in a small flash as the chart updates after loading.  Should we wait until user preferences are retrieved or is this okay?
* Do we want to change the per page options?  Currently set to 1-20. /cc @LevinMedia
